checks.d/yarn.py

```diff
@@ -0,0 +1,264 @@
+    'allocatedMB': ('yarn.apps.allocatedMB', GAUGE),
+    'allocatedVCores': ('yarn.apps.allocatedVCores', GAUGE),
+    'runningContainers': ('yarn.apps.runningContainers', GAUGE),
+    'memorySeconds': ('yarn.apps.memorySeconds', GAUGE),
+    'vcoreSeconds': ('yarn.apps.vcoreSeconds', GAUGE),
+}
+
+YARN_NODE_METRICS = {
+    'lastHealthUpdate': ('yarn.node.lastHealthUpdate', GAUGE),
+    'usedMemoryMB': ('yarn.node.usedMemoryMB', GAUGE),
+    'availMemoryMB': ('yarn.node.availMemoryMB', GAUGE),
+    'usedVirtualCores': ('yarn.node.usedVirtualCores', GAUGE),
+    'availableVirtualCores': ('yarn.node.availableVirtualCores', GAUGE),
+    'numContainers': ('yarn.node.numContainers', GAUGE),
+}
+
+
+class YarnCheck(AgentCheck):
+
+    def check(self, instance):
+
+        # Build the resourceManager URI
+        resourcemanager_hostname = instance.get('resourcemanager_address', DEFAULT_HOSTNAME)
+        resourcemanager_port = instance.get('resourcemanager_port', DEFAULT_RESOURCEMANAGER_PORT)
+        rm_address = resourcemanager_hostname + ':' + str(resourcemanager_port)
+
+        # Get the cluster ID from Yarn
+        cluster_id = self._get_cluster_id(rm_address)
+
+        # If a Cluster ID was not found, assign it 0
+        if not cluster_id:
+            cluster_id = 0
+
+        self._yarn_cluster_metrics(resourcemanager_hostname, cluster_id, rm_address)
+        self._yarn_app_metrics(resourcemanager_hostname, rm_address)
+        self._yarn_node_metrics(resourcemanager_hostname, cluster_id, rm_address)
+
+    def _get_cluster_id(self, rm_address):
+
+        # Get the cluster info
+        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+
+        if info_json:
+            if info_json['clusterInfo']:
+                if info_json['clusterInfo']['id']:
+                    return info_json['clusterInfo']['id']
+
+        return None
+
+
+    def _yarn_cluster_metrics(self, hostname, cluster_id, rm_address):
+        metrics_json = self._rest_request_to_json(rm_address, YARN_CLUSTER_METRICS_PATH)
+
+        if metrics_json:
+            keys = ['cluster_id:' + str(cluster_id)]
+
+            if metrics_json[YARN_CLUSTER_METRICS_ELEMENT] is not None:
+
+                yarn_metrics = metrics_json[YARN_CLUSTER_METRICS_ELEMENT]
+
+                self._set_yarn_metrics_from_json(keys, hostname, yarn_metrics, YARN_CLUSTER_METRICS)
+
+    def _yarn_app_metrics(self, hostname, rm_address):
+        metrics_json = self._rest_request_to_json(rm_address, YARN_APPS_PATH)
+
+        if metrics_json:
+
+            if metrics_json['apps'] is not None:
+                if metrics_json['apps']['app'] is not None:
+
+                    for app_json in metrics_json['apps']['app']:
+
+                        cluster_id = app_json['clusterId']
+                        app_id = app_json['id']
+
+                        keys = ['cluster_id:' + str(cluster_id)]
+                        keys.append('app_id:' + str(app_id))
```

Comment:
Why not adding this tag when instanciating the list ?
Also you should call the list `tags` instead of `keys`
```python
tags = ["cluster_id:%s" % cluster_id, "app_id:%s" % app_id]
```
Als